crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -245,71 +244,35 @@ impl<Provider: ResolverProvider, InstalledPackages: InstalledPackagesProvider>
         let (request_sink, request_stream) = mpsc::channel(300);
 
         // Run the fetcher.
-        let requests_fut = state
-            .clone()
-            .fetch(provider.clone(), request_stream)
-            .map_err(|err| (err, FxHashSet::default()))
-            .fuse();
+        let requests_fut = state.clone().fetch(provider.clone(), request_stream).fuse();
 
         // Spawn the PubGrub solver on a dedicated thread.
         let solver = state.clone();
+        let index_locations = provider.index_locations().clone();
```

Comment:
We have to pass this manually since it's the only thing the solver otherwise doesn't know about when constructing the error.